docs/helm/helm_completion.md

```diff
@@ -29,9 +29,10 @@ helm completion SHELL
       --host string               address of Tiller. Overrides $HELM_HOST
       --kube-context string       name of the kubeconfig context to use
       --tiller-namespace string   namespace of Tiller (default "kube-system")
+      --wait-timeout int          the duration (in seconds) Helm will wait for Tiller pod readiness and status serving (default 300)
```

Comment:
Isn't this going to be confusing since `--wait` for install/upgrade uses `--timeout`?

Reply:
Yeah we can't change the existing --timeout flag, and the two flags address two separate parts of the system. What about `--connection-timeout`, or `--tiller-wait-timeout`? Would those be better descriptors? Any ideas?

Reply:
I suggested `--tiller-wait-timeout` so I bias towards that. It's very descriptive. `--connection-timeout` is not too bad either actually.

Is it going to be a general connection health check that happens to work with `init --wait` or exclusive to `init --wait`? Based on the answer you can pick the more generic or specific.

Reply:
> Is it going to be a general connection health check that happens to work with init --wait or exclusive to init --wait?

The former :) We hook into it and share the same value for `init --wait`, but it's applicable for all commands that require a connection to tiller.

Prior to this PR, the connection timeout was set to 5 seconds, which was *way* too short to wait for tiller to come up and become responsive. Because this feature was broken from the start, we fixed it by bumping the connection timeout up to 5 minutes (similar to #3506), and made it configurable.

I agree that --tiller-wait-timeout or --tiller-connection-timeout might be a more descriptive option and would make it clear how it differentiates from install's --timeout flag. @adamreese any thoughts?